front/lib/api/assistant/agent.ts

```diff
@@ -42,15 +58,103 @@ export async function createAgentConfiguration(
     action?: AgentActionConfigurationType;
     generation?: AgentGenerationConfigurationType;
   }
-): Promise<AgentConfigurationType> {
-  return {
-    sId: generateModelSId(),
-    name,
-    pictureUrl: pictureUrl ?? null,
-    status: "active",
-    action: action ?? null,
-    generation: generation ?? null,
-  };
+): Promise<AgentConfigurationType | void> {
+  const owner = auth.workspace();
+
+  if (!owner) {
```

Comment:
What's the final goal, throw? or support no owner so that we can create system agents? Did we decide to have a status on agent configuration to say whether they are workspace or system in the end?

Reply:
Ah I see we have scope. I presume we would be bumping some of our agent to scope global? So you can probably throw here?